trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogRegister.java

```diff
@@ -0,0 +1,232 @@
+
+    Properties properties = new Properties();
+    properties.put("user", config.getTrinoUser());
+    properties.put("password", config.getTrinoPassword());
+    try {
+      connection = driver.connect(config.getTrinoURI(), properties);
+    } catch (SQLException e) {
+      throw new TrinoException(
+          GRAVITINO_RUNTIME_ERROR, "Failed to initialize Trino the trino connection.", e);
+    }
+
+    catalogStoreDirectory = config.getCatalogStoreDirectory();
+    if (!Files.exists(Path.of(catalogStoreDirectory))) {
+      throw new TrinoException(
+          GRAVITINO_MISSING_CONFIG,
+          "Error config for Trino catalog store directory, file not found");
+    }
+  }
+
+  private String generateCreateCatalogCommand(String name, GravitinoCatalog gravitinoCatalog)
+      throws Exception {
+    return String.format(
+        "CREATE CATALOG %s USING gravitino WITH ( \"%s\" = 'true', \"%s\" = '%s', %s)",
+        name,
+        GRAVITINO_DYNAMIC_CONNECTOR,
+        GRAVITINO_DYNAMIC_CONNECTOR_CATALOG_CONFIG,
+        GravitinoCatalog.toJson(gravitinoCatalog),
+        config.toCatalogConfig());
+  }
+
+  private String generateDropCatalogCommand(String name) throws Exception {
+    return String.format("DROP CATALOG %s", name);
+  }
+
+  void registerCatalog(String name, GravitinoCatalog catalog) {
+    try {
+      String catalogFileName = String.format("%s/%s.properties", catalogStoreDirectory, name);
+      File catalogFile = new File(catalogFileName);
+      if (catalogFile.exists()) {
+        String catalogContents = Files.readString(catalogFile.toPath());
+        if (!catalogContents.contains(GRAVITINO_DYNAMIC_CONNECTOR + "=true")) {
+          throw new TrinoException(
+              GRAVITINO_DUPLICATED_CATALOGS,
+              "Catalog already exists, the catalog is not created by gravitino");
+        } else {
+          throw new TrinoException(
+              GRAVITINO_CATALOG_ALREADY_EXISTS,
+              String.format(
+                  "Catalog %s in metalake %s already exists",
+                  catalog.getName(), catalog.getMetalake()));
+        }
+      }
+
+      if (checkCatalogExist(name)) {
+        throw new TrinoException(
+            GRAVITINO_DUPLICATED_CATALOGS, "Catalog already exists with unknown reason");
+      }
+      String createCatalogCommand = generateCreateCatalogCommand(name, catalog);
+      executeSql(createCatalogCommand);
+      LOG.info("Register catalog {} successfully: {}", name, createCatalogCommand);
+    } catch (Exception e) {
+      String message = String.format("Failed to register catalog %s", name);
+      LOG.error(message);
+      throw new TrinoException(GRAVITINO_RUNTIME_ERROR, message, e);
+    }
+  }
+
+  boolean checkCatalogExist(String name) {
+    String showCatalogCommand = String.format("SHOW CATALOGS like '%s'", name);
+    Exception failedException = null;
+    try {
+      int retries = EXECUTE_QUERY_MAX_RETRIES;
+      while (retries-- > 0) {
```

Comment:
Can use `Awaitility`?

Reply:
I don't want to invoke another thread. It has almost executed successfully.